02-intermediate-ruby/object-inheritance.md

```diff
@@ -3,77 +3,138 @@
 
 ### Using `super` to invoke behavior in _base classes_.
 
-It's also possible to use the behavior defined in the _base class_ alongside specialized behavior in the _child class_. By using the keyword `super`, we can invoke the code in the _base class_ at any point in the _overriding_ method __and__ still be capable of using overriding code.
+It's also possible to use the behavior defined in the _base class_ alongside specialized behavior in the _child class_. By using the keyword `super`, we can invoke the code in the _base class_ at any point in the _overriding_ method __and__ still be capable of using overriding code.  In other words the overriding method can do the same action as the parent class **and** more.
+
+```ruby
+class AudioBook < Book
+  def about
+    # super invokes the `about` method on the _base class_ (Book)
+    return super + "-Audio"
+  end
+end
+
+generic_item = Product.new
+generic_item.about # => "12345"
+
+audiobook = AudioBook.new
+audiobook.about # => "12345: Book-Audio"
+```
+
+In this case, we utilize the behavior that the _base class_ (`Book`) provides but augment it with behavior specific to the _subclass_ (`AudioBook`).
+
+![super in method calls](images/super.png)
+
+## Super & Initialize
+
+We've seen how methods by themselves work with methods, what about instance variables?  Instance variables are inherited as well, for example:
 
 ```ruby
-class Hawk < Bird
-  def talk
-    # super invokes the `talk` method on the _base class_ (Bird)
-    super + " *soars*"
+class Product
+  def initialize(id, name)
+    @id = id
+    @name = name
+  end
+
+  def about
+    return "#{id}: #{name}"
   end
 end
 
-generic_bird = Bird.new
-generic_bird.talk # => "I'm a bird. Kah."
+class Book < Product
+  attr_reader :author, :isbn
 
-hawk = Hawk.new
-hawk.talk # => "I'm a bird. Kah. *soars*"
+  def initialize(id, title, author, isbn)
+    super(id, title)
+    @author = author
+    @isbn = isbn
+  end
+end
 ```
 
-In this case, we utilize the behavior that the _base class_ (`Bird`) provides but augment it with behavior specific to the _subclass_ (`Hawk`).
+In the above example notice the `super` keyword in the 1st line of the `Book` class' `initialize` method.
+
+In initialize, `super` calls the parent, or superclass' constructor.  So `Product`'s `initialize` method is called and `@id` and `@name` are set to the given parameter.
+
+`super` must be the first line in a subclass' `initialize` method.  If `super` is called without an argument, it will use the parameters from the subclass' `initialize` method.
+
+![Super in initialize ](images/inheritance-super.png)
+
+**Exercise**  With your seatmates create an `Video` class.  A `Video` will have, in addition to an ID number and a name, a director and publisher.  It will also have a method which returns the format of the video (BlueRay, DVD, Streaming).  The `about` method should list the format of the video, the ID and the name. Then create an instance of the `Video` class.
+
```

Comment:
Why would format be a method here - seems like it should be a property. I also don't know that making it a method really demonstrates any of the learning goals.